src/thingsboard.c

```diff
@@ -270,6 +355,8 @@ int thingsboard_init(attr_write_callback_t cb, const struct tb_fw_id *fw_id)
 		return ret;
 	}
 
+	k_sem_reset(&rpc_sem);
```

Comment:
Isn't the timestamp already an RPC call? If we decide to use your public API here for out internal stuff. We should not change anything afterwards. I would probable move it more to the top of the init function.